Core/Core/CourseSync/CourseSyncProgress/ViewModel/CourseSyncProgressViewModel.swift

```diff
@@ -65,13 +78,24 @@ class CourseSyncProgressViewModel: ObservableObject {
         self.interactor = interactor
         self.router = router
         updateState(interactor)
-        handleCancelButtonTap(interactor)
+        handleCancelButtonTap()
+        handleCancelConfirmButtonTap(interactor)
         handleDismissButtonTap(interactor)
         handleRetryButtonTap(interactor, router: router)
     }
 
-    private func handleCancelButtonTap(_ interactor: CourseSyncProgressInteractor) {
+    private func handleCancelButtonTap() {
         cancelButtonDidTap
+            .sink { [unowned self] _ in
+                isShowingCancelDialog = true
+            }
+            .store(in: &subscriptions)
+    }
+
+    private func handleCancelConfirmButtonTap(_ interactor: CourseSyncProgressInteractor) {
+        confirmAlert
+            .userConfirmation()
```

Comment:
This will finish the stream after the first cancel/confirmation, so a Cancel - No - Cancel - Yes flow won't cancel the sync. The intended way to use this is to chain this call in a `flatMap` after we received a `cancelButtonDidTap` event.

Reply:
I missed the function description, changed the logic.